src/main/java/uk/gov/pay/ledger/transaction/model/CsvTransactionFactory.java

```diff
@@ -194,7 +194,11 @@ public CsvTransactionFactory(ObjectMapper objectMapper) {
         }
 
         if (metadataKeys != null) {
-            metadataKeys.stream().sorted()
+            metadataKeys
+                    .stream()
+                    .map(String::toLowerCase)
```

Comment:
I think this will change all metadata headers to always be in lowercase. Could impact services if their processing is case sensitive. 

Suggested change would be to add to headers like below, so in case on duplicate keys last one will take preference. Otherwise, columns stay unchanged.

`headers.put(header.toLowerCase(), header);`

Reply:
I see why this would be beneficial for existing integrations, the problem for the future is that it will be unpredictable -- it would be difficult to make any assertions about what to expect headers to be called if there is a conflict

Reply:
I see. This could be sensitive change in that case. A way could be, if there is a duplicate, it will always be lowered. If not it will be presented as is